README.md

```diff
@@ -1,2 +1,351 @@
+
+To access any contained methods:
+new.insert(val)
+new.search(val)
+new.size()
+new.depth(root)
+new.contains(val)
+new.balance()
+new.in_order()
+new.pre_order()
+new.post_order()
+new.breadth_first()
+new.delete(val)
+```
+
+### Hash Table
+
+```
+Using a naive hash and slightly more complex hash with Horner's Rule, HashTable() will store based on the supplied key, which must be a string. By default, HashTable() uses the naive hash method, but can be changed on initialization. 
+
+To create an instance if the hash table, HashTable(), from python:
+
+new = HashTable() *you may initiate the table with any size, must be integer, as the first arguement in the function. You may also pass in an alternative hashing method. A naive hash will be used my defaul, but you may also choose `horner_hash` as the second arguement on initialization, or any hashing function you choose to import. For example: HashTable(100000, horner_hash).*
+
+HashTable() contains the following methods:
+* _set(key, val) (O(n))_ - sets a value at a given key (must me string) in the table, if key already has value it will be set to new value.
+* _get(key) (0(n))_ - retrieve the value of the item with the given key.
+* _ _hash(key) (O(n))_ - hashes key and returns an integer between 0 and the size of the table.
+
+To access any contained methods:
+new.set(key, val)
+new.get(key)
+```
+
+### Trie
+
+```
+In order to efficiently store words, Trie() will provide a class by which nodes are only inserted if unshared by anthor word. All words share previous characters when possible. 
+
+To create an instance if the trie, Trie(), from python:
+
+new = Trie() *you may not initiate the trie with any values.*
+
+Trie() contains the following methods:
+* _insert(string) (O log(n))_ - inserts a new node for every letter in string (input must be word) if not able to share with another word.
+* _contains(string) (0 log(n))_ - returns True if word is in trie and False if not.
+* _size() (O(1))_ - retruns the number of words in the trie.
+* _remove(string) (O log(n))_ - removes word from trie and raises error if word not in trie.
+
+To access any contained methods:
+new.insert(string)
+new.contains(string)
+new.size()
+new.remove(string)
+```
+
+### Bubble Sort
+
+```
+Bubble sort takes in a list of numbers and uses a bubble sort method to return a sorted list. 
+
+To use bubble_sort, from bubble_sort import bubble_sort.
+Pass in a list of numbers bubble_sort(list).
+
+* _bubble_sort(list) (O(n^2))_
+
+```
+
+### Insert Sort
+
+```
+Insert sort takes in a list of numbers and uses an insert sort method to return a sorted list. 
+
+To use insert_sort, from insert_sort import insert_sort.
+Pass in a list of numbers insert_sort(list).
+
+* _insert_sort(list) (O(n^2))_
+
+```
+
+### Merge Sort
+
+```
+Merge sort takes in a list of numbers and uses an merge sort method to return a sorted list. 
+
+To use merge_sort, from merge_sort import merge_sort.
```

Comment:
Probably put back-ticks around the import statement here to further distinguish the code that someone would write from the text that's explaining what's happening